crates/holochain_conductor_api/src/config/conductor/dpki_config.rs

```diff
@@ -41,27 +38,25 @@ pub struct DpkiConfig {
     pub allow_throwaway_random_dpki_agent_key: bool,
 
     /// For testing only, we can turn off DPKI if needed.
-    /// TODO: this can be removed once DPKI is truly optional again.
+    // TODO: this can be removed once DPKI is truly optional again.
     #[serde(default)]
     pub no_dpki: bool,
 }
 
 impl DpkiConfig {
-    pub fn production(dna_path: Option<PathBuf>) -> Self {
-        Self {
-            dna_path,
-            network_seed: DPKI_NETWORK_SEED_MAIN.to_string(),
-            allow_throwaway_random_dpki_agent_key: false,
-            no_dpki: false,
-        }
-    }
-
     pub fn testing() -> Self {
-        Self {
-            dna_path: None,
-            network_seed: DPKI_NETWORK_SEED_TESTING.to_string(),
-            allow_throwaway_random_dpki_agent_key: true,
-            no_dpki: false,
+        cfg_if! {
+            if #[cfg(feature = "unstable-dpki")] {
+                Self {
+                    dna_path: None,
+                    network_seed: DPKI_NETWORK_SEED_TESTING.to_string(),
+                    allow_throwaway_random_dpki_agent_key: true,
+                    no_dpki: false,
+                }
+            } else {
+                tracing::error!("Enabling DPKI on conductor without specifying cargo feature 'unstable-dpki' at compile time.");
```

Comment:
If unstable-dpki feature is not set, but DPKI is explicitly configured for testing, an error is logged.